src/ray/raylet/task_dependency_manager.cc

```diff
@@ -297,6 +298,44 @@ void TaskDependencyManager::TaskCanceled(const TaskID &task_id) {
   }
 }
 
+void TaskDependencyManager::CleanupForDriver(const std::unordered_set<TaskID> &task_ids) {
+  for (auto it = task_dependencies_.begin(); it != task_dependencies_.end();) {
+    if (task_ids.find(it->first) != task_ids.end()) {
+      it = task_dependencies_.erase(it);
+    } else {
+      it++;
+    }
+  }
+
+  for (auto it = required_tasks_.begin(); it != required_tasks_.end();) {
+    if (task_ids.find(it->first) != task_ids.end()) {
+      it = required_tasks_.erase(it);
+    } else {
+      it++;
+    }
+  }
+
+  for (auto it = pending_tasks_.begin(); it != pending_tasks_.end();) {
+    if (task_ids.find(it->first) != task_ids.end()) {
+      it = pending_tasks_.erase(it);
+    } else {
+      it++;
+    }
+  }
```

Comment:
I think it'd be better to loop over `task_ids`, because
1) `task_ids` should be smaller than those three maps, so perf will be better (N*logM vs M*logN).
2) code will be cleaner
```
for (auto it = task_ids.begin(); it != task_ids.end(); it++) {
  task_dependencies_.erase(*it);
  required_tasks_.erase(*it);
  pending_tasks_.erase(*it);
```

Reply:
makes sense. thanks!